src/components/Grid.tsx

```diff
@@ -0,0 +1,118 @@
+import type { ComponentPropsWithoutRef, ElementType } from 'react';
+
+import { css } from '@emotion/react';
+
+interface GridProps extends ComponentPropsWithoutRef<ElementType> {
+  /**
+   * Grid 컴포넌트가 사용할 HTML 태그
+   *
+   * @default 'div'
+   */
+  tag?: ElementType;
+  /**
+   * Grid 컴포넌트의 행의 크기를 지정하는 방법
+   */
+  rows?: string | number;
+  /**
+   * Grid 컴포넌트의 열의 크기를 지정하는 방법
+   */
+  columns?: string | number;
+  /**
+   * Grid 컴포넌트 내 아이템들 사이의 행 간격
+   *
+   * @default 'normal'
+   */
+  rowGap?: string | number;
+  /**
+   * Grid 컴포넌트 내 아이템들 사이의 열 간격
+   *
+   * @default 'normal'
+   */
+  columnGap?: string | number;
+  /**
+   * Grid 컴포넌트 내 아이템들 사이의 간격
+   *
+   * @default 'normal'
+   */
+  gap?: string | number;
+}
+
+function Grid({
+  tag = 'div',
+  rows,
+  columns,
+  rowGap,
+  columnGap,
+  gap,
+  children,
+  ...attributes
+}: GridProps) {
+  const Tag = tag;
+
+  const gridTemplateRows = String(rows)?.match(/^\d+$/) ? `repeat(${rows}, minmax(0, 1fr))` : rows;
+  const gridTemplateColumns = String(columns)?.match(/^\d+$/)
+    ? `repeat(${columns}, minmax(0, 1fr))`
+    : columns;
```

Comment:
`rows`, `columns`의 값에 따라 속성이 결정되는 로직이네요! 멋집니다 👍

추가로 `minmax`을 두신 이유가 궁금해요! 저는 `1fr`만 있어도 충분할 것이라고 생각하는데, 이렇게 된다면 어떤 문제가 있다고 생각하신건가요?

Reply:
#### `repeat(3, 1fr)`
<img width="1728" alt="Screenshot 2023-09-19 at 4 59 20 PM" src="https://github.com/woowacourse/layout-component/assets/51967731/d963a38a-4096-4acd-a554-c0b0cfe9c6b3">

<img width="87" alt="Screenshot 2023-09-19 at 5 42 41 PM" src="https://github.com/woowacourse/layout-component/assets/51967731/ad7a046b-7c0a-4ed5-b4ce-1955f00405d3">


#### `repeat(3, minmax(0, 1fr))`
<img width="1728" alt="Screenshot 2023-09-19 at 4 59 34 PM" src="https://github.com/woowacourse/layout-component/assets/51967731/e0c98dd6-f27b-4186-9e5d-2a9660ee47c5">

<img width="88" alt="Screenshot 2023-09-19 at 5 42 29 PM" src="https://github.com/woowacourse/layout-component/assets/51967731/3b6950f9-6215-4360-9ce5-2ea03c517b02">


<br>
<br>

> minmax(0,1fr) ensures content overflows in the child instead of the parent when there is not enough available space for each fraction to be at least the size of the largest item.

https://stackoverflow.com/questions/73543647/do-we-need-minmax-in-repeat

`1fr`로 설정하면 화면 너비가 위에 이미지처럼 엄청 줄었을 때 그리드의 자식 요소가 자기의 기본/최소 너비까지만 줄어요. 그런데 `minmax(0, 1fr)`로 설정하면 그리드 요소 간격/너비(?) 자체가 0으로 줄어들어요. 그래서 위에 이미지처럼 collapse 하게 돼요. 만약 자식 요소가 `padding`이나 `width`가 없으면 `1fr`과 `minmax(0, 1fr)`이 거의 비슷하게 작동하는 것 같아요. 그런데 반응형을 생각했을 때 화면 너비, 즉, 그리드 너비가 줄었을 때 자식들도 그에 맞춰서 변화하면 좋지 않을까라는 생각이 들었어요! 그런데 실제로 사용자 입장에서는 디폴트로 자식의 기본 너비를 유지하는 것이 좋은지 아니면 `minmax(0, 1fr)`처럼 보이게 하는게 더 좋은지는 모르겠네요...🤔